### PR TITLE
[Communication] - Phone Numbers - Wait on sync LRO calls for samples and tests

### PR DESCRIPTION
**Pull request by minnieliu**

### CODE REVIEW

File: sdk/communication/azure-communication-phonenumbers/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Purchased phone numbers can come with many capabilities, depending on the countr
 
 ### Initializing Phone Number Client
 The PhoneNumberClientBuilder is enabled to use Azure Active Directory Authentication
-<!-- embedme ./src/samples/java/com/azure/communication/phonenumbers/ReadmeSamples.java#L52-L62 -->
+<!-- embedme ./src/samples/java/com/azure/communication/phonenumbers/ReadmeSamples.java#L54-L64 -->
```
Comment (sacheu) [inline sdk/communication/azure-communication-phonenumbers/README.md:33]: How to verify these line numbers are correct?

File: sdk/communication/azure-communication-phonenumbers/README.md [hunk 2]
```diff
@@ -103,36 +103,41 @@ The Phone Number Client supports a variety of long running operations that allow
...
 
 ### Purchase Phone Numbers
 The result of searching for phone numbers is a `PhoneNumberSearchResult`. This can be used to get the numbers' details and purchase numbers by passing in the `searchId` to the purchase number API.
 
-<!-- embedme ./src/samples/java/com/azure/communication/phonenumbers/ReadmeSamples.java#L111-L113 -->
+<!-- embedme ./src/samples/java/com/azure/communication/phonenumbers/ReadmeSamples.java#L118-L120 -->
 ```java
 PollResponse<PhoneNumberOperation> purchaseResponse =
-    phoneNumberClient.beginPurchasePhoneNumbers(searchResult.getSearchId(), Context.NONE).waitForCompletion();
+    phoneNumberClient.beginPurchasePhoneNumbers(searchId, Context.NONE).waitForCompletion();
```
Comment (sacheu) [inline sdk/communication/azure-communication-phonenumbers/README.md:133]: Why we need to change this? This seem unrelated to this PR.

File: sdk/communication/azure-communication-phonenumbers/README.md [hunk 3]
```diff
@@ -103,36 +103,41 @@ The Phone Number Client supports a variety of long running operations that allow
...
-    .beginSearchAvailablePhoneNumbers("US", PhoneNumberType.TOLL_FREE, PhoneNumberAssignmentType.APPLICATION, capabilities, searchOptions, Context.NONE)
-    .getFinalResult();
-
-System.out.println("Searched phone numbers: " + searchResult.getPhoneNumbers());
-System.out.println("Search expires by: " + searchResult.getSearchExpiresBy());
-System.out.println("Phone number costs:" + searchResult.getCost().getAmount());
+SyncPoller<PhoneNumberOperation, PhoneNumberSearchResult> poller = phoneNumberClient
+    .beginSearchAvailablePhoneNumbers("US", PhoneNumberType.TOLL_FREE, PhoneNumberAssignmentType.APPLICATION, capabilities, searchOptions, Context.NONE);
+PollResponse<PhoneNumberOperation> response = poller.waitForCompletion();
+String searchId = "";
```
Comment (sacheu) [inline sdk/communication/azure-communication-phonenumbers/README.md:116]: nit: Can this line combine with line 120?

File: sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java [hunk 4]
```diff
@@ -104,12 +111,18 @@ public void beginSearchAvailablePhoneNumbers(HttpClient httpClient) {
...
+        PollResponse<PhoneNumberOperation> response = poller.waitForCompletion();
+        if (LongRunningOperationStatus.SUCCESSFULLY_COMPLETED == response.getStatus()) {
+            PhoneNumberSearchResult searchResult = poller.getFinalResult();
+            String phoneNumber = getTestPhoneNumber(searchResult.getPhoneNumbers().get(0));
+            PollResponse<PhoneNumberOperation> purchaseOperationResponse = beginPurchasePhoneNumbersHelper(httpClient, searchResult.getSearchId(), "beginPurchasePhoneNumbersWithoutContextSync", false).waitForCompletion();
+            assertEquals(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, purchaseOperationResponse.getStatus());
+            PollResponse<PhoneNumberOperation> releaseOperationResponse = beginReleasePhoneNumberHelper(httpClient, phoneNumber, "beginReleasePhoneNumberWithoutContextSync", false).waitForCompletion();
+            assertEquals(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, releaseOperationResponse.getStatus());
+        } else {
+            fail("Long Running Operation Status was not successfully completed");
```
Comment (sacheu) [inline sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java:124]: Why use fail() here but assertEquals in line 119 , 121?

File: sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java [hunk 5]
```diff
@@ -118,12 +131,18 @@ public void beginPurchaseandReleasePhoneNumbersWithoutContext(HttpClient httpCli
...
+        PollResponse<PhoneNumberOperation> response = poller.waitForCompletion();
+        if (LongRunningOperationStatus.SUCCESSFULLY_COMPLETED == response.getStatus()) {
+            PhoneNumberSearchResult searchResult = poller.getFinalResult();
+            String phoneNumber = getTestPhoneNumber(searchResult.getPhoneNumbers().get(0));
+            PollResponse<PhoneNumberOperation> purchaseOperationResponse = beginPurchasePhoneNumbersHelper(httpClient, searchResult.getSearchId(), "beginPurchasePhoneNumbersSync", true).waitForCompletion();
+            assertEquals(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, purchaseOperationResponse.getStatus());
+            PollResponse<PhoneNumberOperation> releaseOperationResponse = beginReleasePhoneNumberHelper(httpClient, phoneNumber, "beginReleasePhoneNumberSync", true).waitForCompletion();
+            assertEquals(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, releaseOperationResponse.getStatus());
+        } else {
+            fail("Long Running Operation Status was not successfully completed");
```
Comment (sacheu) [inline sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java:144]: Why use fail() here but assertEquals in line 119 , 121?

**Review by sacheu — COMMENTED** [review]
Looks good Minnie. Just have a few questions.

File: sdk/communication/azure-communication-phonenumbers/README.md (hunk 1 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-phonenumbers/README.md:33]: `npx embedme README.md` populate the code, these snippets are not manually added.

File: sdk/communication/azure-communication-phonenumbers/README.md (hunk 2 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-phonenumbers/README.md:133]: This code comes from the samples, which is updated

File: sdk/communication/azure-communication-phonenumbers/README.md (hunk 3 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-phonenumbers/README.md:116]: No, because we want to use in purchase, which is outside the if statement

File: sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java (hunk 4 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java:124]: fail() is the function we use that would just automatically fail the test unconditionally. https://javacodex.com/JUnit/JUnit-Fail-Method

File: sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java (hunk 5 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java:144]: See above

File: sdk/communication/azure-communication-phonenumbers/README.md (hunk 3 above)
Comment (sacheu) [reply sdk/communication/azure-communication-phonenumbers/README.md:116]: searchId is populated inside the if statement, when it is used outside,  should it have a null check? 
since searchId can be null if the if checks return false.

**Review by sacheu — APPROVED** [review]
Looks Good. Thanks for fixing this.
